admiral/pkg/clusters/serviceentry.go

```diff
@@ -213,7 +213,7 @@ func modifyServiceEntryForNewServiceOrPod(
 			ctxLogger.Infof(common.CtxLogFormat, "event", "", "", clusterId, "neither deployment nor rollouts found")
 			continue
 		}
-		ingressEndpoint, port := getIngressEndpointAndPort(rc)
+		ingressEndpoint, port := rc.ServiceController.Cache.GetLoadBalancer(common.GetAdmiralParams().LabelSet.GatewayApp, common.NamespaceIstioSystem)
```

Comment:
any reason we pulled the code out of the func?

Reply:
No place is using that except this place.
Plan to clean up that method as well. Also func is not doing anything part from sending same thing.